R/utils.R

```diff
@@ -460,13 +457,11 @@
            "Please check that 'sep' is correct.", call. = FALSE)
     }
     taxa_tab <- DataFrame(as.matrix(taxa_split))
-    colnames(taxa_tab) <- all_ranks
+    colnames(taxa_tab) <- names(all_ranks)
     
-    # Subset columns so that it includes TAXONOMY_RANKS columns by default.
-    # If strain column has values, it it also returned.
-    ind <- !(!tolower(colnames(taxa_tab)) %in% tolower(returned.ranks) &
-        colSums(is.na(taxa_tab)) == nrow(taxa_tab))
-    taxa_tab <- taxa_tab[ , ind, drop = FALSE]
+    # Subset columns so that include only those columns that have some value
+    non_empty <- colSums(is.na(taxa_tab)) != nrow(taxa_tab)
+    taxa_tab <- taxa_tab[ , non_empty, drop = FALSE]
     
```

Comment:
The output had "all" the taxonomy ranks even though the data did not include them. Now only those ranks are returned that were detected from the data.